### PR TITLE
ESlint whitespace rules

### PR DESCRIPTION
**Pull request by orangejenny**
https://github.com/dimagi/commcare-hq/pull/21748#discussion_r215267380

When there were multiple options, I erred on the side of matching python conventions and/or picking the style that's already more prevalent in our code (those two things never conflicted, our python habits are strong).

@dannyroberts / @pr33thi

### CODE REVIEW

**Comment by orangejenny** [comment]
(ignoring stickler)

File: .eslintrc.js [hunk 1]
```diff
@@ -44,10 +44,19 @@ module.exports = {
...
+        "no-irregular-whitespace": ["error"],
         "no-new-object": ["error"],
-        "no-unneeded-ternary": ["error"],
+        "no-regex-spaces": ["error"],
+        "no-spaced-func": ["error"],
         "no-throw-literal": ["error"],
-        "no-implicit-globals": ["error"],
+        "no-unneeded-ternary": ["error"],
+        "no-whitespace-before-property": ["error"], // match flake8 E201 and E211
+        "space-before-function-paren": ["error", "never"],
```
Comment (dannyroberts) [inline .eslintrc.js]: I think the rule we have been following (and what I prefer and picked up from crockford style in javascript the good parts when we started trying to have conventions) is `"space-before-function-paren": ["error", {"anonymous": "always", "named": "never", "asyncArrow": "always"}]` meaning

```
function () { // space before anonymous function paren
}
```
but
```
function foo() {  // no space before named function paren
}
```

File: .eslintrc.js [hunk 2]
```diff
@@ -44,10 +44,19 @@ module.exports = {
         "indent": ["warn", 4, {"SwitchCase":1}],
         "linebreak-style": ["error", "unix"],
         "semi": ["error", "always"],
+        "keyword-spacing": ["error"],
+        "no-implicit-globals": ["error"],
+        "no-irregular-whitespace": ["error"],
         "no-new-object": ["error"],
-        "no-unneeded-ternary": ["error"],
+        "no-regex-spaces": ["error"],
+        "no-spaced-func": ["error"],
```
Comment (dannyroberts) [inline .eslintrc.js]: Looks like this has been deprecated in favor of
```
"func-call-spacing": ["error", "never"]
```
according to https://eslint.org/docs/rules/no-spaced-func#disallow-spacing-between-function-identifiers-and-their-applications-no-spaced-func. (Though what's here still works on the current version and does the same thing.)

File: .eslintrc.js [hunk 3]
```diff
@@ -44,10 +44,19 @@ module.exports = {
...
         "no-throw-literal": ["error"],
-        "no-implicit-globals": ["error"],
+        "no-unneeded-ternary": ["error"],
+        "no-whitespace-before-property": ["error"], // match flake8 E201 and E211
+        "space-before-function-paren": ["error", "never"],
+        "space-before-blocks": ["error"],
+        "space-in-parens": ["error", "never"],
+        "space-infix-ops": ["error"],   // match flake8 E225
```
Comment (dannyroberts) [inline .eslintrc.js]: Along these lines, might want to add [key-spacing](https://eslint.org/docs/rules/key-spacing); the defaults seem good so I think we just want `"key-spacing": ["error"]`.

**Review by dannyroberts — COMMENTED** [review]
Made a couple suggestions. I think the `space-before-function-paren` change I suggest will also make this a much, much smaller PR 😄

**Comment by dannyroberts** [comment]
Also, this is great. Thanks for leading the way on this!

**Review by pr33thi — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (orangejenny) [reply .eslintrc.js]: (offlined, going to implement Danny's suggestion)

**Comment by orangejenny** [comment]
@dannyroberts updated

**Review by dannyroberts — APPROVED** [review]
Nice, looks good!

As a final and totally optional note, with the commit history as-is, this will change the git-blame on many lines that ended up being unchanged in this PR (i.e. every line that was previously contained `function ()`, which ultimately remained the same, was still touched by 2b41da9 and 510386e). To get around that, you could `git rebase master -i` and delete the `--fix` commits, and then upon that new history do a final eslint --fix run. If keeping a clean git-blame on these lines doesn't seem important, then probably not worth the trouble!

**Review by dannyroberts — APPROVED** [review]

**Pull request merged**